doc/docs/Python_Tutorials/Basics.md

```diff
@@ -896,6 +896,111 @@ For `resolution = 25`, the error decreases (as expected) to 1.5%.
+                     center=mp.Vector3(-0.5*s+dpml),
+                     size=mp.Vector3(0,s),
+                     component=mp.Ez)]
+
+symmetries = [mp.Mirror(mp.Y)]
+
+geometry = [mp.Cylinder(material=SiO2,
+                        center=mp.Vector3(),
+                        radius=r,
+                        height=mp.inf)]
+
+sim = mp.Simulation(resolution=resolution,
+                    cell_size=cell_size,
+                    boundary_layers=pml_layers,
+                    sources=sources,
+                    k_point=mp.Vector3(),
+                    symmetries=symmetries,
+                    geometry=geometry)
+
+dft_fields = sim.add_dft_fields([mp.Dz,mp.Ez], fcen, 0, 1, center=mp.Vector3(), size=mp.Vector3(2*r,2*r))
+
+# closed box surrounding cylinder for computing total incoming flux
+flux_box = sim.add_flux(fcen, 0, 1,
+                        mp.FluxRegion(center=mp.Vector3(x=-r),size=mp.Vector3(0,2*r),weight=+1),
+                        mp.FluxRegion(center=mp.Vector3(x=+r),size=mp.Vector3(0,2*r),weight=-1),
+                        mp.FluxRegion(center=mp.Vector3(y=+r),size=mp.Vector3(2*r,0),weight=-1),
+                        mp.FluxRegion(center=mp.Vector3(y=-r),size=mp.Vector3(2*r,0),weight=+1))
+
+sim.run(until_after_sources=100)
+
+plt.figure()
+sim.plot2D()
+plt.savefig('power_density_cell.png',dpi=150,bbox_inches='tight')
+
+(x,y,z,w) = sim.get_array_metadata(dft_cell=dft_fields)
+Dz = sim.get_dft_array(dft_fields,mp.Dz,0)
+Ez = sim.get_dft_array(dft_fields,mp.Ez,0)
+absorbed_power_density = 2*np.pi*fcen * np.imag(np.conj(Ez)*Dz)
+
+plt.figure()
+plt.pcolormesh(x,y,np.transpose(absorbed_power_density),cmap='inferno_r',shading='gouraud',vmin=0,vmax=np.amax(absorbed_power_density))
+plt.xlabel("x")
+plt.ylabel("y")
+plt.gca().set_aspect('equal')
+plt.title("absorbed power density")
+plt.colorbar()
+plt.savefig('power_density_map.png',dpi=150,bbox_inches='tight')
+
+absorbed_power = np.sum(w*absorbed_power_density)
+absorbed_flux = mp.get_fluxes(flux_box)[0]
+err = abs(absorbed_power-absorbed_flux)/absorbed_flux
+print("flux:, {} (dft_fields), {} (dft_flux), {} (error)".format(absorbed_power,absorbed_flux,err))
+```
+
+A schematic of the simulation layout generated using [`plot2D`](Python_User_Interface.md#data-visualization) shows the line source (red), PMLs (green hatch region), `dft_flux` box (solid blue contour line), and `dft_fields` surface (blue hatch region).
+
+<center>
+![](../images/power_density_cell.png)
+</center>
+
+The spatial map of the absorbed power density shows that, as expected, most of the absorption occurs on the surface of the cylinder facing the incident planewave. Note that the labeled $x$ and $y$ coordinate axes of the 2d region are obtained using [`get_array_metadata`](Python_User_Interface.md#array-metadata) which also provides the interpolation weights (`w`) used to compute the integral of the absorbed power density.
```

Comment:
It seems to be absorbed in a pretty thin layer.   What is the absorption length λ/Im[sqrt(ε)] at this wavelength for this material?   If the absorption length is comparable to a pixel, maybe you should change the material or the wavelength so that the absorption length is a bit larger and easier to resolve in the plot?

Reply:
It's unclear why the absorption is confined to such a thin layer on the front surface. The pixel width for a resolution of 50 is 0.02 μm and is much smaller than the absorption length (see below).

SiO<sub>2</sub> at λ=1 μm has an absorption length of ~3000 μm (i.e., 3000X the radius of the cylinder).
```py
>>> from meep.materials import SiO2
>>> SiO2.epsilon(1.0)[0][0]
(2.132084213667966+0.0009852435550391251j)
>>> 1.0/np.imag(np.sqrt(SiO2.epsilon(1.0)[0][0]))
2964.070860833652
```

![power_density_map_SiO2_wvl1 0](https://user-images.githubusercontent.com/7152530/86862968-1c83ca00-c07f-11ea-92bd-25e038fbe9ab.png)

By comparison, [ITO](https://github.com/NanoComp/meep/blob/master/python/materials.py#L67-L80) at λ=0.3 μm has a much smaller absorption length (Labs) of ~4 μm.
```py
>>> from meep.materials import ITO
>>> ITO.epsilon(1/0.3)[0][0]
(4.932295718818409+0.3550131776059032j)
>>> 0.3/np.imag(np.sqrt(ITO.epsilon(1/0.3)[0][0]))
3.7558827876760446
```

The spatial map for the ITO cylinder shows that the absorption is more strongly confined to its front surface than that of SiO2.

![power_density_map_ITO_wvl0 3](https://user-images.githubusercontent.com/7152530/86862827-d6c70180-c07e-11ea-830d-7fca58d9cbdd.png)